tests/rptest/tests/compatibility/kafka_streams_test.py

```diff
@@ -387,3 +393,208 @@ def __init__(self, test_context):
+        # Create the two examples needed
+        # 7070 and 7071 are port numbers
+        example1 = self.create_example(False, 7070)
+        example2 = self.create_example(False, 7071)
+
+        # Create the driver
+        driver = self.create_driver(True, None)
+
+        # Start the examples and wait for them to load
+        example1.start()
+        example2.start()
+        example1.wait()
+        example2.wait()
+
+        # Start the driver and wait for it to load
+        driver.start()
+        driver.wait()
+
+        host = example1.node_name()
+
+        # Do HTTP GET on the possible routes
+        self._http_get(host, "/state/instances",
+                       lambda text: "windowed-word-count" in text)
+
+        self._http_get(host, "/state/instances/word-count",
+                       lambda text: "word-count" in text)
+
+        self._http_get(host, "/state/instance/word-count/hello",
+                       lambda text: "word-count" in text)
+
+        self._http_get(host, "/state/keyvalue/word-count/hello",
+                       lambda text: "hello" in text)
+
+        # HTTP GET on /state/keyvalues/word-count/all often
+        # responds with 500 error.
+
+        driver.stop()
+        example2.stop()
+        example1.stop()
+
+
+# Not using any KafkaStreams bases classes here because this
+# test runs a single process in the background (a.k.a BackgroundThreadService)
+# but the other tests run multiple background processes.
+class KafkaStreamsMicroservices(KafkaStreamsTest):
+    """
+    Test KafkaStreams Microservices examples which demonstrate an order management
+    workflow, such as one might find in retail and online shopping.
+    """
+
+    topics = (
+        TopicSpec(name="orders"),
+        TopicSpec(name="warehouse-inventory"),
+        TopicSpec(name="order-validations"),
+        TopicSpec(name="payments"),
+        TopicSpec(name="customers"),
+        TopicSpec(name="platinum"),
+        TopicSpec(name="gold"),
+        TopicSpec(name="silver"),
+        TopicSpec(name="bronze"),
+    )
+
+    Example = KafkaStreamExamples.KafkaStreamsMicroservices
+
+    def __init__(self, test_context):
+        super(KafkaStreamsMicroservices,
+              self).__init__(test_context=test_context,
+                             enable_pp=False,
+                             enable_sr=False)
+
+    @cluster(num_nodes=4)
+    @matrix(jar=[
+        "EmailServiceTest", "FraudServiceTest", "InventoryServiceTest",
+        "OrderDetailsServiceTest", "OrdersServiceTest",
+        "ValidationsAggregatorServiceTest"
+    ])
+    def test_kafka_streams_microservices(self, jar):
+        example = self.create_example(jar)
+        example.start()
+        example.wait()
+        example.stop()
+
+
+# Not using any of the KafkaStreams base classes because this test
+# runs a client-side script to "reset" the app in the middle of execution
+# that other tests do not use.
+class KafkaStreamsAppReset(KafkaStreamsTest):
```

Comment:
It's better not to reuse names because it makes harder to search and to navigate through the code base. Here we have `KafkaStreamsAppReset` used in two places: in this file and in `KafkaStreamExamples.KafkaStreamsAppReset`

Reply:
I'll make the change.